### ISSUE TITLE
Files larger than 4MB cannot be downloaded

### ISSUE THREAD

**Issue by vsosinga** [umbraco-community/UmbracoFileSystemProviders.Azure#158]
Files larger than 4MB cannot be download from Umbraco. When opening a large file it seems to be cut off resulting in a corrupt download.

To reproduce:
1. Create an empty Umbraco project
2. Add UmbracoFileSystemProviders.Azure & UmbracoFileSystemProviders.Azure.Media v2.0.0-beta1
3. Configure a default Azure blob storage
4. Upload a file larger than 4MB
5. Try to download the uploaded file

Uploaded file:
![image](https://user-images.githubusercontent.com/38254373/74666095-304a9480-51a1-11ea-98cf-fef5df6fbe58.png)

In the blob storage:
![image](https://user-images.githubusercontent.com/38254373/74666307-8cadb400-51a1-11ea-9f85-aed0f216bc35.png)

Downloaded file:
![image](https://user-images.githubusercontent.com/38254373/74666076-24f76900-51a1-11ea-911e-33f54fe2eba9.png)

This issue seems to be introduced with pull request https://github.com/umbraco-community/UmbracoFileSystemProviders.Azure/pull/156 .
The same test case on v2.0.0-alpha5 gives no problems.

**Comment by Jeavon** [comment]
Thanks @vsosinga I'll investigate this

**Comment by timeverts** [comment]
I can confirm the same issue here running UmbracoFileSystemProviders.Azure v1.1.0. Files larger than 4MB seem to upload fine, but do not download beyond 4MB of data. This is even visible in the CMS.

![image](https://user-images.githubusercontent.com/15150632/75735395-87ba3a00-5d35-11ea-8ab4-3e811a894d80.png)

If I hit the URL of the image directly in the blob, the full image loads fine.

I believe the same issue is also affecting PDF files larger than 4MB. The PDF uploads fine and can be read directly from the blob storage URL, but when trying to open the PDF from the Umbraco URL, a 'Failed to load PDF document.' is returned.

![image](https://user-images.githubusercontent.com/15150632/75735508-cc45d580-5d35-11ea-959f-d2e7744522c7.png)

Hope you can resolve this soon, as it is preventing us from migrating to Azure Blob storage for several of the Umbraco 7 sites of our clients.

**Comment by c9mbundy** [comment]
FWIW - I am running UmbracoFileSystemProviders.Azure 1.0.3 with Umbraco 7 (Umbraco-Cloud) and downloading 28MB video files without issue - perhaps in the recent update?

**Comment by timeverts** [comment]
@c9mbundy, @Jeavon, I just downgraded to v1.0.3 with Umbraco 7 (native) and can also confirm that the bug is not present in that version. So just summarizing, UmbracoFileSystemProviders.Azure v1.0.3 works fine, but UmbracoFileSystemProviders.Azure v1.1.0 does not.

I did some more investigating of the differences between the two versions and found that there was a [commit made for v1.1.0](https://github.com/umbraco-community/UmbracoFileSystemProviders.Azure/commit/365db1445ac55b2af44cffab323a930c1d642068) that changed the way files are read, using the [CloudBlob.OpenRead()](https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.storage.blob.cloudblob.openread) method rather than streaming them from memory. From doing some Googling, I think that in certain configurations this method has a limit of 4MB of read data (see [this example](https://stackoverflow.com/questions/6911728/cloudblob-openread-is-not-reading-all-data)) and reading files larger than that may need to use multiple appended chunks.

So @Jeavon, I reckon this is a good starting point for getting this problem rectified. Hope you're able to resolve soon.
